File: server/src/main/java/com/example/schoolsite/entity/Classroom.java
package com.example.schoolsite.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "classroom")
@ToString
public class Classroom {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @Column(name = "classroomTeacherId", nullable = false)
    private Long classroomTeacherId; // TODO: сделать внешним ключом
    @Column(name = "name", nullable = false)
    private String name;

    public Classroom(Long classroomTeacherId, String name) {
        this.classroomTeacherId = classroomTeacherId;
        this.name = name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Classroom classroom = (Classroom) o;
        return Objects.equals(id, classroom.id) && Objects.equals(classroomTeacherId, classroom.classroomTeacherId) && Objects.equals(name, classroom.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, classroomTeacherId, name);
    }
}

File: server/src/main/java/com/example/schoolsite/entity/Classroom.kt
package com.example.schoolsite.entity

import lombok.Getter

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "classroom")
@ToString
class Classroom(classroomTeacherId: Teacher, name: String) {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private val id: Long? = null

    @OneToOne(optional = false, cascade = CascadeType.ALL)
    @JoinColumn(name = "classroom_teacher_id", nullable = false)
    private val classroomTeacherId: Teacher

    @Column(name = "name", nullable = false)
    private val name: String

    @OneToOne(optional = false, mappedBy = "classroomId")
    private val pupil: Pupil? = null

    @OneToOne(optional = false, mappedBy = "classroomID")
    private val shedule: Shedule? = null

    @OneToOne(optional = false, mappedBy = "classID")
    private val attendance: Attendance? = null

    @OneToOne(optional = false, mappedBy = "classID")
    private val academicPerfomance: AcademicPerfomance? = null

    init {
        this.classroomTeacherId = classroomTeacherId
        this.name = name
    }

    @Override
    override fun equals(o: Object?): Boolean {
        if (this === o) return true
        if (o == null || getClass() !== o.getClass()) return false
        val classroom = o as Classroom
        return Objects.equals(id, classroom.id) && Objects.equals(classroomTeacherId, classroom.classroomTeacherId) && Objects.equals(name, classroom.name)
    }

    @Override
    override fun hashCode(): Int {
        return Objects.hash(id, classroomTeacherId, name)
    }
}